### PR TITLE
Support for dumping DateTime and Dates

### PR DESCRIPTION
**Pull request by ilikeorangutans**
The dumper currently only formats Time instances according to the TOML
spec. However, some ruby programs use DateTimes or Dates which should
also be formatted according to RFC3339.

DateTimes are formatted in the same way as Date instances. Dates are
formatted `yyyy-mm-dd` without a timezone offset as ruby dates do not
hold an offset

This PR adds `require 'date`' where needed. I'm not sure if this is acceptable in this project, but `date` is part of the rubylib and should be fine on any installation.

### CODE REVIEW

**Comment by emancu** [comment]
@ilikeorangutans Thanks for your PR. I'm out on vacations now, I will take a look as soon as possible. I need to review the other PR too, but this one will be easier for sure!

**Comment by emancu** [comment]
Also I need to review the latest version of TOML and specs.

**Comment by ilikeorangutans** [comment]
No worries; I have a hack in my project that works around, but this would much cleaner. Enjoy your vacation and let me know if you need help with any of this.

**Comment by emancu** [comment]
I didn't forget about this PR. I am working on the support for Toml 0.5 which is taking longer than expected.

I am still not sure if I want to introduce a dependency that could fire me back and make backwards compatibility harder on future implementations. However, I'm not discarding this either since toml-0.5 introduces some changes on date/timestamp types.

Hope to have updates soon!

File: test/dumper_test.rb [hunk 1]
```diff
@@ -28,8 +29,14 @@ def test_dump_types
     dumped = TomlRB.dump(array: [[1, 2], %w(weird one)])
     assert_equal("array = [[1, 2], [\"weird\", \"one\"]]\n", dumped)
 
-    dumped = TomlRB.dump(datetime: Time.utc(1986, 8, 28, 15, 15))
+    dumped = TomlRB.dump(time: Time.utc(1986, 8, 28, 15, 15))
+    assert_equal("time = 1986-08-28T15:15:00Z\n", dumped)
+
+    dumped = TomlRB.dump(datetime: DateTime.new(1986, 8, 28, 15, 15))
```
[the remark this answers is not in the document]
Comment (emancu) [reply test/dumper_test.rb]: I know, but 🤷‍♂

**Comment by emancu** [comment]
@ilikeorangutans Great PR, I'm merging an releasing now!

File: test/dumper_test.rb [hunk 2]
```diff
@@ -28,9 +29,15 @@ def test_dump_types
...
     assert_equal("datetime = 1986-08-28T15:15:00Z\n", dumped)
 
+    dumped = TomlRB.dump(date: Date.new(1986, 8, 28))
+    assert_equal("datetime = 1986-08-28\n", dumped)
```
Comment (emancu) [inline test/dumper_test.rb]:
```suggestion
    assert_equal("date = 1986-08-28\n", dumped)
```

**Comment by ilikeorangutans** [comment]
Horray! Thanks for merging! 🎉

**Pull request merged**